src/python/pants/jvm/run.py

```diff
@@ -174,6 +180,24 @@ async def create_run_request(
     return _post_process_jvm_process(proc, jdk)
 
 
+@memoized
+def _jvm_source_run_request_rule(cls: type[JvmRunnableSourceFieldSet]) -> Iterable[Rule]:
+    @rule(
+        canonical_name_suffix=cls.__name__,
+        _param_type_overrides={"request": cls},
+        level=LogLevel.DEBUG,
+    )
+    async def jvm_source_run_request(request: JvmRunnableSourceFieldSet) -> RunRequest:
+        return await create_run_request(GenericJvmRunRequest(request), **implicitly())
+
+    return [*run_rules(), *collect_rules(locals())]
+
+
+def jvm_rules(cls: type[JvmRunnableSourceFieldSet]) -> Iterable[Rule | UnionRule]:
```

Comment:
Maybe call this `jvm_run_rules_for_field_set` or something similar to convey to the reader what it does (now that it is standalone).

Reply:
I can add `_run_` but I don't see the need to mention the type, since it's in the signature already.